launch_frontend/launch_frontend/action_parse_methods.py

```diff
@@ -0,0 +1,126 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Module for launch action parsing methods."""
+
+import shlex
+
+import launch
+from launch.conditions import IfCondition
+from launch.conditions import UnlessCondition
+from launch.substitutions import TextSubstitution
+
+from .entity import Entity
+from .expose import expose_action
+from .parser import Parser
+
+
+def parse_action(entity: Entity, parser: Parser):
+    """
+    Parse action.
+
+    This is only intented for code reusage, and it's not exposed.
+    """
+    if_cond = entity.get_attr('if', optional=True)
+    unless_cond = entity.get_attr('unless', optional=True)
+    kwargs = {}
+    if if_cond is not None and unless_cond is not None:
+        raise RuntimeError("if and unless conditions can't be used simultaneously")
+    if if_cond is not None:
+        kwargs['condition'] = IfCondition(predicate_expression=if_cond)
+    if unless_cond is not None:
+        kwargs['condition'] = UnlessCondition(predicate_expression=unless_cond)
+    return launch.Action, kwargs
+
+
+@expose_action('executable')
+def parse_executable(
+    entity: Entity,
+    parser: Parser,
+    optional_cmd: bool = False
+):
+    """
+    Parse executable tag.
+
+    :param: optional_cmd Allow not specifying `cmd` argument.
+        Intended for code reuse in derived classes (e.g.: launch_ros.actions.Node).
+    """
+    cmd = entity.get_attr('cmd', optional=optional_cmd)
+    if cmd is not None:
+        cmd_list = [parser.parse_substitution(cmd)]
+    else:
+        cmd_list = []
+    kwargs = {}
+    cwd = entity.get_attr('cwd', optional=True)
+    if cwd is not None:
+        kwargs['cwd'] = parser.parse_substitution(cwd)
+    name = entity.get_attr('name', optional=True)
+    if name is not None:
+        kwargs['name'] = parser.parse_substitution(name)
+    prefix = entity.get_attr('launch-prefix', optional=True)
+    if prefix is not None:
+        kwargs['prefix'] = parser.parse_substitution(prefix)
+    output = entity.get_attr('output', optional=True)
+    if output is not None:
+        kwargs['output'] = output
+    shell = entity.get_attr('shell', types='bool', optional=True)
+    if shell is not None:
+        kwargs['shell'] = shell
+    # Conditions won't be allowed in the `env` tag.
+    # If that feature is needed, `set_enviroment_variable` and
+    # `unset_enviroment_variable` actions should be used.
+    env = entity.get_attr('env', types='list[Entity]', optional=True)
+    if env is not None:
+        env = {e.get_attr('name'): parser.parse_substitution(e.get_attr('value')) for e in env}
+        kwargs['additional_env'] = env
+    args = entity.get_attr('args', optional=True)
+    # `args` is supposed to be a list separated with ' '.
+    # All the found `TextSubstitution` items are split and
+    # added to the list again as a `TextSubstitution`.
+    # TODO(ivanpauno): Change `ExecuteProcess` api from accepting
```

Comment:
@ivanpauno same as the other TODO, I think we should file this or it'll get lost.